pkg/hnsw/heap.go

```diff
@@ -0,0 +1,115 @@
+package hnsw
+
+import (
+	"fmt"
+)
+
+type Item struct {
+	id   Id
+	dist float32
+}
+
+var EmptyHeapError = fmt.Errorf("Empty Heap")
+
+type HeapInterface interface {
+	PeekMinItem() (*Item, error)
+	PeekMaxItem() (*Item, error)
+	PopMinItem() (*Item, error)
+	PopMaxItem() (*Item, error)
+	Insert(id Id, dist float32)
+	IsEmpty() bool
+	Len() int
+	Less(i, j int) bool
+	Swap(i, j int)
+	Push(x *Item)
+	Pop() *Item
+}
```

Comment:
It would also be fine to not even have this interface and make the functions in `minmax.go` class functions which would remove the need for this file all together. The only reason `heap.Interface` exists is to allow external implementers but we won't have any of those.